app/controllers/creative_previews_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CreativePreviewsController < ApplicationController
+  before_action :authenticate_user!
+
+  include AdRenderable
```

Comment:
I'd move AdRenderable above the `before_action` to be the first line inside the class def.